src/features/products/productsSlice.ts

```diff
@@ -28,13 +49,52 @@ export const productsSlice = createSlice({
   initialState,
   reducers: {
     changeSearch(state, action: PayloadAction<string>) {
-      state.searchValue = action.payload;
+      state.filter.searchValue = action.payload;
     },
     clearSearch(state) {
-      state.searchValue = "";
+      state.filter.searchValue = "";
     },
     changeCategory(state, action: PayloadAction<string>) {
-      state.category = action.payload;
+      state.filter.category = action.payload;
+      state.filter.brands = [];
+    },
+    changeSingleBrand(state, action: PayloadAction<string>) {
+      state.filter.category = "";
+      state.filter.stars = [];
+      state.filter.price = [state.minMaxPrice.min, state.minMaxPrice.max];
+      state.filter.brands = [action.payload];
+    },
+    toggleBrands(state, action: PayloadAction<string>) {
+      if (state.filter.brands.includes(action.payload)) {
+        const filteredBrands = state.filter.brands.filter(
+          (brand) => brand !== action.payload
+        );
+        state.filter.brands = filteredBrands;
+      } else {
+        state.filter.brands.push(action.payload);
+      }
+    },
+    toggleStars(state, action: PayloadAction<number>) {
+      if (state.filter.stars.includes(action.payload)) {
+        const filteredStars = state.filter.stars.filter(
+          (star) => star !== action.payload
+        );
+        state.filter.stars = filteredStars;
+      } else {
+        state.filter.stars.push(action.payload);
+      }
+    },
```

Comment:
Try to simplify

Reply:
Fixed - create universal toggle function